### ISSUE TITLE
ENH: make bec smarter (tell the difference between detectors and motors)

### ISSUE THREAD

**Issue by ambarb** [NSLS-II/bluesky#949]
If bec could tell the difference between detectors and motors then most of the situations like that described in #948 would be taken care of automatically.

THere was some discussion with @tacaswell about creating a griddle plot, with the columns of plots belonging to the same motor scan.  THis would be also appreciated for things line `inner_product_scan()` for things like theta-2theta scans and HKL scans.  See the plot below for an `inner_product scan()` utlizing the motors assciated with HKL.  Please note that `tardis` is also in the list of detectors (it needs to be so that the physical motors are recorderd in the database -  but these are not needed to make decisions with the plot).

![inner_product_hkl_scan_plot](https://user-images.githubusercontent.com/13630233/34881890-d277b146-f782-11e7-9b44-52c9476ed481.png)

**Comment by teddyrendahl** [comment]
Could we do this with a convention in `hints`? Right now I think the current convention is 
```python
{'fields': ['readback', 'setpoint']}
```
but we could maybe have `fields` and `measurements` (just throwing a name out there), for separating between `hints` that are typically on the horizontal axis and those that are usually on the vertical? Just spit-balling.

**Comment by tacaswell** [comment]
IPS needs to provide better hints / we need to document what hints are being used.

**Comment by danielballan** [comment]
This distinction is one that hints aims to capture already -- but it evidently isn't implemented well yet in the case of `tardis`. Recall that hints come from two places: plans, which stash a 'hints' field in the RunStart document, and Devices, which stash a 'hints' field in the EventDescriptor document(s). The distinction between "vertical" (y) and "horizonal" (x) is best specified by the _plan_, not the Device, because it knows whether a Device is being used as a motor or a detector. A given Device could serve either role; the plan is in a better position to know which is applicable in a given run.

**Comment by jrmlhermitte** [comment]
My guess here is that @ambarb was scanning `tardis_h`, `tardis_k` and `tardis_l`, correct?
This is a bug. BEC should be smart enough to know motors are independent axes. See #963.

**Comment by jrmlhermitte** [comment]
PS: The convention should also be:
```py
hints = {'fields' : ['fccd1_stats1_total'],
              'dimensions' : [ (['tardis_h'], 'primary'),]
              }
```
if you wanted to scan on `tardis_h`. Note that the bug would still produce two extra plots on `tardis_k` and `tardis_l`. A fix needs to be merged before we have any control over plotting like this (without mangling with the descriptor data, i.e. removing `tardis_h` and `tardis_l` as motors...)

Note : the api for hints is not set in stone yet.

**Comment by ambarb** [comment]
Yes, this was `relative_inner_product_scan` for motors `tardis.h`, `tardis.k` and `tardis.l`

We also include the detector list to be: `[fccd, tardis]`.  This is because we want the diffractometer motor positions recorded (i.e., `tardis.theta`) as well as the reciprocal lattice dimensions in the event of a single axis scan. 

It would be nice to have 3 plots (horizontally maybe) with 3 different x axes.  Could this be done automatically within `bec`?

**Comment by stan-dot** [comment]
> It would be nice to have 3 plots (horizontally maybe) with 3 different x axes. Could this be done automatically within bec?

this does look interesting
